src/Declaration.js

```diff
@@ -48,8 +54,10 @@ export default class Declaration {
 	run ( strongDependencies ) {
 		if ( this.tested ) return this.hasSideEffects;
 
-
-		if ( !this.functionNode ) {
+		if ( this.isClassDeclaration && !this.hasConstructor ) {
+			// class declaration without a constructor has no side-effects
+			this.hasSideEffects = false;
```

Comment:
A second condition for this to be true is for no _super class constructor_ had side-effects, as a constructor-less class still implicitly invokes the constructor of its super class.